elements/pfe-clipboard/test/pfe-clipboard.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect, aTimeout, oneEvent, html } from '@open-wc/testing';
+import { spy, stub, SinonStub } from 'sinon';
+import { createFixture } from '@patternfly/pfe-tools/test/create-fixture.js';
+import { hexToRgb } from '@patternfly/pfe-tools/test/hex-to-rgb.js';
+import { PfeClipboard } from '@patternfly/pfe-clipboard';
+
+const element = html`
+  <pfe-clipboard></pfe-clipboard>
+`;
+
+describe('<pfe-clipboard>', async function() {
+  const clipboardMocks = new Set<SinonStub>();
+
+  beforeEach(() => clipboardMocks.add(stub(navigator.clipboard, 'writeText')));
+  afterEach(() => clipboardMocks.forEach(x => x.restore()));
+
+  it('should upgrade', async function() {
+    const el = await createFixture<PfeClipboard>(element);
+    expect(el)
+      .to.be.an.instanceOf(customElements.get('pfe-clipboard'))
+      .and
+      .to.be.an.instanceOf(PfeClipboard);
+  });
+
+  it('should render the default slot content.', async function() {
+    const el = await createFixture<PfeClipboard>(element);
+    expect(el.shadowRoot!.querySelector('#text')!.textContent).to.equal('Copy URL');
+    expect(el.shadowRoot!.querySelector('#text--success')!.textContent).to.equal('Copied');
+    expect(el.shadowRoot!.querySelector<HTMLSlotElement>(`#icon`)!.assignedElements().length).to.equal(0);
+  });
+
+  it('it should render slot overrides', async function() {
+    const el = await createFixture<PfeClipboard>(element);
+    // The default slot override will be handled by transposeSlot
+    const defaultSlot = `<span slot="text">You can totally click to copy url</span>`;
+    const textSuccessSlot = `<span slot="text--success">Making some copies!</span>`;
+    const iconSlot = `<pfe-icon slot="icon" icon="web-icon-globe" color="darker"></pfe-icon>`;
+    el.innerHTML = `
+        ${defaultSlot}
+        ${textSuccessSlot}
+        ${iconSlot}
+    `;
+    // transposeSlot should have sent it to the text named slot
+    expect(el.shadowRoot!.querySelector<HTMLSlotElement>(`#text`)!.assignedElements({ flatten: true })
+      .map(i => i.outerHTML.trim()).join('')).to.equal(defaultSlot);
+    // The success and icon slots should be working as expected also
+    expect(el.shadowRoot!.querySelector<HTMLSlotElement>(`#text--success`)!.assignedElements({ flatten: true })
+      .map(i => i.outerHTML.trim()).join('')).to.equal(textSuccessSlot);
+    // deprecated slot
+    expect(el.shadowRoot!.querySelector<HTMLSlotElement>(`#text--success`)!.assignedElements({ flatten: true })
```

Comment:
This test will fail if we stack the the `success-text` and `text--success` slots.  It can probably be removed or a new test will need to be written to test that the `text--success` slot still works.

Reply:
please check again

Reply:
👍